src/js/services/auth.js

```diff
@@ -108,17 +108,18 @@ export default App.extend({
   },
   initBootstrap() {
     const d = $.Deferred();
-    $.when(Radio.request('entities', 'fetch:clinicians:current')).then(currentUser => {
+    const fetchCurrentUser = Radio.request('entities', 'fetch:clinicians:current');
+    const fetchGroups = Radio.request('entities', 'fetch:groups:collection');
+    const fetchRoles = Radio.request('entities', 'fetch:roles:collection');
+    const fetchStates = Radio.request('entities', 'fetch:states:collection');
+    $.when(fetchCurrentUser, fetchRoles, fetchStates, fetchGroups).done((currentUser, roles, states) => {
       this.currentUser = currentUser;
-      this.currentOrg = this.currentUser.getOrganization();
-
-      const groupRequests = this.currentOrg.getGroups().map(group => {
-        return Radio.request('entities', 'fetch:clinicians:byGroup', group);
-      });
-
-      $.when(...groupRequests).done(() => {
-        d.resolve(currentUser);
+      this.currentOrg = Radio.request('entities', 'organizations:model', {
+        states: states[0],
+        roles: roles[0],
```

Comment:
can you explain this for me? This is getting the cached organization (which is the default, for now) based on the roles/states that come back from the api?

Reply:
This is just creating an organization from nowhere. it's not cached, but it has a default value set for the name.  Then I'm attaching states and roles mainly because it's convenient, requires the fewest changes, and largely still makes sense from the FE architecture's perspective.  We could attach states and roles here like we're attaching `currentUser` and `currentOrg` and add 2 additional auth requests.  I was on the fence.  The roles, states, and groups can still be considered the org ones..  though at the moment we only need to request groups related to the currentUser (though we need all groups cached for various display)  thoughts?

Reply:
🤔 it makes sense as these three models are owned by the org (and, in the case of roles/states will eventually be customized by them). I think the difficulty lies in just changing the way I think about "Orgs" in the context of the application, and less in how it's architected if that makes sense. 

On the other hand, the frontend application will only ever get back roles/states for the 'current' org because of sso/auth magic right? So does accessing them through the org make sense there? Or should they be their own, standalone, thing? I dunno, I'm fine either way.

Reply:
the roles, states and groups will always be the same for the subdomain.  The organization is essentially non-existent, but for the frontend it's a name we throw on a model to display on the top left menu.. so since it's an entity for FE.. it is effectively equivalent to the subdomain which owns the other entities.  organization for FE is to a certain extent "global"